tests/foreman/api/test_ansible.py

```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
     )
     assert task_details[0].result == 'success'
     assert len(task_details[0].output['result']['created']) == playbooks_count
+
+
+def test_positive_assign_ansible_role(target_sat):
+    """
+    Test API for assigning an Ansible Role to a host
+
+    :id: 95d0213e-a5a9-4671-8a9e-89a77fc67f39
+
+    :customerscenario: false
+
+    :steps:
+
+        1. Create a Host
+        2. Import a roles into satellite
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
+    host_roles = host.list_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
```

Comment:
If we have a list of all roles, we could turn most of this into a fixture and parametrize the test with the list of roles.

Reply:
Hmm, I do like that idea
This is just the first of the available roles on any given satellite. Maybe make a constant list of those, and then do as you suggest